packages/visual-editor/src/components/pageSections/NearbyLocations.tsx

```diff
@@ -297,11 +302,24 @@ const LocationCard = ({
   address: AddressType;
   timezone: string;
   mainPhone: string;
-  id: string;
-  slug?: string;
+  locationData: any;
+  puck: PuckContext;
 }) => {
-  const { relativePrefixToRoot } = useTemplateProps();
-  const { i18n } = useTranslation();
+  const { document: streamDocument, relativePrefixToRoot } = useTemplateProps();
+
+  const documentForUrlTemplate = {
+    ...locationData,
```

Comment:
This isn't using the right nearby location data now. It's using the current page document.